pkg/packetfilter/nftables/nftables.go

```diff
@@ -0,0 +1,535 @@
+}
+
+func (p *packetFilter) DeleteIPHookChain(chain *packetfilter.ChainIPHook) error {
+	return p.deleteChain(chain.Name)
+}
+
+func (p *packetFilter) DeleteChain(_ packetfilter.TableType, chain string) error {
+	return p.deleteChain(chain)
+}
+
+func (p *packetFilter) deleteChain(chain string) error {
+	tx := p.nftables.NewTransaction()
+	tx.Delete(&knftables.Chain{
+		Name: chain,
+	})
+
+	err := p.nftables.Run(context.TODO(), tx)
+	if knftables.IsNotFound(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "error deleting chain %q", chain)
+}
+
+func (p *packetFilter) ClearChain(_ packetfilter.TableType, chain string) error {
+	tx := p.nftables.NewTransaction()
+	tx.Flush(&knftables.Chain{
+		Name: chain,
+	})
+
+	err := p.nftables.Run(context.TODO(), tx)
+	if knftables.IsNotFound(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "error clearing chain %q", chain)
+}
+
+func (p *packetFilter) Delete(_ packetfilter.TableType, chain string, rule *packetfilter.Rule) error {
+	kRule := p.getNftablesRuleFromList(chain, rule)
+	if kRule == nil {
+		return nil
+	}
+
+	tx := p.nftables.NewTransaction()
+	tx.Delete(&knftables.Rule{
+		Chain:  chain,
+		Handle: kRule.Handle,
+	})
+
+	err := p.nftables.Run(context.TODO(), tx)
+	if knftables.IsNotFound(err) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "error deleting rule %q from chain %q", *kRule.Comment, chain)
+}
+
+func (p *packetFilter) getNftablesRuleFromList(chain string, rule *packetfilter.Rule) *knftables.Rule {
+	existingRules, err := p.nftables.ListRules(context.TODO(), chain)
+	if err != nil {
+		return nil
+	}
+
+	for _, existingRule := range existingRules {
+		if ToRuleSpec(rule).String() == *existingRule.Comment {
+			return existingRule
+		}
+	}
+
+	return nil
+}
+
+func (p *packetFilter) List(_ packetfilter.TableType, chain string) ([]*packetfilter.Rule, error) {
+	existingRules, err := p.nftables.ListRules(context.TODO(), chain)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list rules of chain %q", chain)
+	}
+
+	rules := []*packetfilter.Rule{}
+
+	// Note: since currently ListRules returns only `Comment` and `Handle` fields
+	// of Rule object. When creating a new rule `Comment` will be set to RuleSpec.
+
+	for _, existingRule := range existingRules {
+		ruleSpec := strings.Split(*existingRule.Comment, " ")
+		rules = append(rules, FromRuleSpec(ruleSpec))
+	}
+
+	return rules, nil
+}
+
+func (p *packetFilter) Insert(_ packetfilter.TableType, chain string, pos int, rule *packetfilter.Rule) error {
+	return p.insertRuleAtPosition(chain, rule, pos)
+}
+
+func (p *packetFilter) Append(_ packetfilter.TableType, chain string, rule *packetfilter.Rule) error {
+	return p.insertRuleAtPosition(chain, rule, -1)
+}
+
+func (p *packetFilter) insertRuleAtPosition(chain string, rule *packetfilter.Rule, pos int) error {
+	knftRule := knftables.Rule{
+		Chain:   chain,
+		Rule:    ToRuleSpec(rule).String(),
+		Comment: ptr.To(ToRuleSpec(rule).String()),
+	}
```

Comment:
We should only call `ToRuleSpec` once. 

```suggestion
        ruleSpec := ToRuleSpec(rule).String()
        
	knftRule := knftables.Rule{
		Chain:   chain,
		Rule:    ruleSpec,
		Comment: ptr.To(ruleSpec),
	}
```